### PR TITLE
GG-34108 Fixed REST authentication failure

### PR DESCRIPTION
**Pull request by sk0x50**
https://ggsystems.atlassian.net/browse/GG-34108

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java [hunk 1]
```diff
@@ -152,4 +179,61 @@ public void testAdditionalPasswordServerFail() throws Exception {
...
+
+        assertTrue("Failed to wait for starting node.", startLatch.await(10, SECONDS));
+
+        try (GridClient client = GridClientFactory.start(getGridClientConfiguration())) {
+            err.set(client.checkLastError());
+        }
+
+        contLatch.countDown();
+
+        startFut.get();
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java:233]: Please add timeout

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java [hunk 2]
```diff
@@ -152,4 +179,61 @@ public void testAdditionalPasswordServerFail() throws Exception {
...
+
+        Exception unexpectedErr = err.get();
+
+        assertNull("Unexpected error [err=" + unexpectedErr + ']', unexpectedErr);
```
Comment (tkalkirill) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java:237]:
```suggestion
        Exception unexpectedErr = err.get();

        assertNull(err.get());
```

**Review by tkalkirill — APPROVED** [review]

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java (hunk 2 above)
Comment (sk0x50) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java:237]: I don't like assertions without a message. On the other hand, in general, we should avoid reading err atomics twice. So, I would like to stay with the current code.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java (hunk 1 above)
Comment (sk0x50) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/security/client/AdditionalSecurityCheckTest.java:233]: Yep, will fix it.

**Pull request merged**